src/utils/apr.ts

```diff
@@ -27,15 +27,16 @@ export const getPoolApr = (
  * @param poolWeight allocationPoint / totalAllocationPoint
  * @param cakePriceUsd Cake price in USD
  * @param poolLiquidityUsd Total pool liquidity in USD
- * @returns
+ * @param farmAddress Farm Address
+ * @returns Farm Apr
  */
 export const getFarmApr = (
   poolWeight: BigNumber,
   cakePriceUsd: BigNumber,
   poolLiquidityUsd: BigNumber,
   farmAddress: string,
 ): { cakeRewardsApr: number; lpRewardsApr: number } => {
-  const yearlyCakeRewardAllocation = CAKE_PER_YEAR.times(poolWeight)
+  const yearlyCakeRewardAllocation = poolWeight ? poolWeight.times(CAKE_PER_YEAR) : new BigNumber(NaN)
```

Comment:
Why `NaN`? Wouldn't it be better to have it be 0?

Reply:
If poolWeight is null and others not, (although it is not possible because in https://github.com/pancakeswap/pancake-frontend/blob/363b7400c168097585df14ed35bed97de96e60b2/src/state/farms/fetchPublicFarmData.ts#L90 we put zero if we don't have totalAllocPoint) , 

if we assign 0 to yearlyCakeRewardAllocation, cakeRewardsApr will be 0 instead of null that is why I choose NaN there.